jscomp/core/lam_compile.ml

```diff
@@ -512,9 +515,20 @@ and compile_general_cases
           [switch ?default ?declaration switch_exp body]
         )
 
-and compile_cases cxt switch_exp table default =
+and compile_cases ~is_tag cxt switch_exp table default sw_names =
```

Comment:
can we change it to a more meaningful name like `is_nullary`

Reply:
when `sw_names` is None, `is_tag` does not make sense here, can we change it to just one extract argument of type `((_ -> _) * bool) option`

Reply:
cleaned up the code